homeassistant/components/netatmo/sensor.py

```diff
@@ -144,18 +166,7 @@ def _retry(_data):
             if CONF_MODULES in config:
                 module_items = config[CONF_MODULES]
                 for module_name in module_items:
-                    if module_name not in data.get_module_names():
-                        continue
-                    for condition in data.station_data.monitoredConditions(
-                            module_name):
-                        dev.append(
-                            NetatmoSensor(
-                                data,
-                                module_name,
-                                condition.lower(),
-                                data.station
-                            )
-                        )
+                    dev.extend(find_devices(data, [module_name]))
```

Comment:
Hmm, you didn't remove the loop. I meant to simplify like this:
```python
    if CONF_MODULES in config:
        dev.extend(find_devices(data, config[CONF_MODULES]))
```

(you might even be able to combine the "manual" and "all" cases now since `config.get(CONF_MODULES)` will give the needed value for the second argument in both cases ... but maybe that is too cute)